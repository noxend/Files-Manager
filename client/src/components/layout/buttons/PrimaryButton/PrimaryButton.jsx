import React from 'react';
import { Button } from 'antd';

const PrimaryButton = (props) => {
  return (
    <Button type="primary" {...props}>
      {props.children}
    </Button>
  );
};

export default PrimaryButton;
