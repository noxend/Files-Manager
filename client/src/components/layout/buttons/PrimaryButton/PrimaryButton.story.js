import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';

import PrimaryButton from './PrimaryButton';

const stories = storiesOf('Buttons', module);

stories.addDecorator(withKnobs);

stories.add('Primary Button', () => {
  const name = text('Button Name', 'Some name');

  const loading = boolean('Loading', false);
  const disabled = boolean('Disabled', false);

  return (
    <PrimaryButton loading={loading} disabled={disabled}>
      {name}
    </PrimaryButton>
  );
});
