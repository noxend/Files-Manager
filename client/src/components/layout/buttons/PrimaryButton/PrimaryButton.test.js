import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PrimaryButton from './PrimaryButton';

configure({ adapter: new Adapter() });

describe('Default Button', () => {
  it('Snapshot', () => {
    const tree = renderer.create(<PrimaryButton>Title</PrimaryButton>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Props should to match the component', () => {
    const wrapper = shallow(<PrimaryButton>Title</PrimaryButton>);
    expect(wrapper.find('Button').prop('type')).toBe('primary');
    expect(wrapper.find('Button').prop('children')).toBe('Title');
  });
});
