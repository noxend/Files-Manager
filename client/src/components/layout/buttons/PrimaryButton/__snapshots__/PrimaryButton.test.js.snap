// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Button Snapshot 1`] = `
<button
  className="ant-btn ant-btn-primary"
  onClick={[Function]}
  type="button"
>
  <span>
    Title
  </span>
</button>
`;
